python/graphstorm/inference/utils.py

```diff
@@ -0,0 +1,89 @@
+    """ A utility function to zip model artifacts into a tar package
+
+    This function will put the entry point file into a sub-folder,named `code`, and then zip the
+    `model.bin`, the `**.yaml`, the `**.json` file, and the `code` sub-folder into a tar package
+    with the name specified by the wrap_name argument, and save it to the `output_path`.
+
+    This function will check if given files exist. Non-existance will trigger an error raising.
+    
+    Parameters
+    ----------
+    path_to_model: str
+        The path of GraphStorm's `model.bin` file.
+    path_to_yaml: str
+        The path of the YAML file generated by GraphStorm during model training.
+    path_to_json: str
+        The path of JSON file generated by GraphStorm gconstruct or GSProcessing.
+    path_to_entry: str
+        The path of the entry point file for a specific task. The file will be put into a
+        sub-folder, named 'code'.
+    wrap_name: str
+        The name of the tar package. Default is `model`.
+    output_path: str
+        The folder where the output tar package will be saved. If not provided, will
+        raise an error. Default is None, meaning not provided.
+
+    """
+    # check if files exist
+    assert os.path.exists(path_to_model), f'The model file, {path_to_model}, does not exist.'
+    assert os.path.isfile(path_to_model), f'The {path_to_model} should be a file path, \
+                                            but got a folder.'
+    assert os.path.exists(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           does not exist.'
+    assert os.path.isfile(path_to_yaml), f'The {path_to_yaml} should be a file path, \
+                                           but got a folder.'
+    assert os.path.exists(path_to_json), f'The graph metadata JSON file, {path_to_json}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_json), f'The {path_to_json} should be a file path, \
+                                           but got a folder.'
+    assert os.path.exists(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_entry), f'The {path_to_entry} should be a file path, \
+                                            but got a folder.'
+
+    # create the output folder if not exist.
+    assert output_path is not None, 'Output path is not provided.'
+    assert os.path.isdir(output_path), 'Output path should be a folder name, but got a ' + \
+                                       f'file {output_path}.'
+    os.makedirs(output_path, exist_ok=True)
+    os.makedirs(os.path.join(output_path, 'code'), exist_ok=True)
+
+    # copy the artifacts files to output_path
+    shutil.copy(path_to_entry, os.path.join(output_path, 'code', os.path.basename(path_to_entry)))
+    if output_path != os.path.dirname(path_to_model):
```

Comment:
output_path is a directory and path_to_model is a file path. The if statement will always be True.

Are you check whether path_to_model has already been under output_path? 
You may want to use `os.path.dirname`